pkg/config/engine/api.go

```diff
@@ -23,4 +23,31 @@ type Interface interface {
 	Set(string, interface{})
 	RemoveRuntime(string) error
 	Save(string) (int64, error)
+	GetRuntimeConfig(string) (RuntimeConfig, error)
+}
+
+// RuntimeConfig defines the interface to query container runtime handler configuration
+type RuntimeConfig interface {
+	GetBinaryPath() string
+}
+
+func GetBinaryPathsForRuntimes(cfg Interface, runtimes ...string) []string {
+	var binaryPaths []string
+
+	seen := make(map[string]bool)
+	for _, runtime := range runtimes {
+		runtimeConfig, err := cfg.GetRuntimeConfig(runtime)
+		if err != nil {
+			// TODO: It will be useful to log the error when GetRuntimeConfig fails for a runtime
+			continue
+		}
+		binaryPath := runtimeConfig.GetBinaryPath()
+		if binaryPath == "" || seen[binaryPath] {
+			continue
+		}
+		seen[binaryPath] = true
+		binaryPaths = append(binaryPaths, binaryPath)
+	}
+
+	return binaryPaths
```

Comment:
```suggestion
// GetBinaryPathsForRuntimes returns the list of binary paths for common rutimes.
// An optional list of runtimes is checked in addition to the following list:
//
//	the default runtime, "runc", and "crun"
//
// If an nvidia* runtime is set as the default runtime, this is ignored.
func GetBinaryPathsForRuntimes(cfg Interface, runtimes ...string) []string {
	isValidDefault := func(s string) bool {
		if s == "" {
			return false
		}
		// ignore nvidia* runtimes.
		return !strings.HasPrefix(s, "nvidia")
	}
	if defaultRuntime := cfg.DefaultRuntime(); isValidDefault(defaultRuntime) {
		runtimes = append(runtimes, defaultRuntime)
	}

	runtimes = append(runtimes, "runc", "crun")

	var binaryPaths []string
	seen := make(map[string]bool)
	for _, runtime := range runtimes {
		runtimeConfig, err := cfg.GetRuntimeConfig(runtime)
		if err != nil {
			// TODO: It will be useful to log the error when GetRuntimeConfig fails for a runtime
			continue
		}
		binaryPath := runtimeConfig.GetBinaryPath()
		if binaryPath == "" || seen[binaryPath] {
			continue
		}
		seen[binaryPath] = true
		binaryPaths = append(binaryPaths, binaryPath)
	}

	return binaryPaths
}
```